### PR TITLE
Add a basic integration test

### PR DESCRIPTION
**Pull request by hammerhead**
## Summary of the changes / Why this is an improvement

* Update dependencies
* Add a basic Terratest smoke test for AWS/Azure (no automatic GitHub action for now)
* Add dependabot for Go dependencies

## Checklist

 - [X] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: test/cratedb.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	"errors"
+	"context"
+)
+
+func RunCrateDBQuery(clusterUrl string, username string, password string) (map[string]interface{}, error) {
+	client := retryablehttp.NewClient()
+	client.HTTPClient.Transport = &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client.RetryWaitMin = 5 * time.Minute
```
[the remark this answers is not in the document]
Comment (hammerhead) [reply test/cratedb.go]: The effect of `RetryWaitMin` is very unclear, also from the documentation of go-retryablehttp.
From running the test, one can see that the wait time between two tries is actually 30 seconds:
```
2021/12/21 11:39:57 [DEBUG] POST https://20.126.152.193:4200/_sql
2021/12/21 11:41:12 [ERR] POST https://20.126.152.193:4200/_sql request failed: Post "https://20.126.152.193:4200/_sql": dial tcp 20.126.152.193:4200: connect: operation timed out
2021/12/21 11:41:12 [DEBUG] POST https://20.126.152.193:4200/_sql: retrying in 30s (40 left)
```
I've removed the setting, it uses a dynamic backoff algorithm then also reaching 30 seconds quickly:
```
2021/12/21 12:01:54 [DEBUG] POST https://13.94.148.187:4200/_sql
2021/12/21 12:02:13 [DEBUG] POST https://13.94.148.187:4200/_sql (status: 401): retrying in 1s (20 left)
2021/12/21 12:02:14 [DEBUG] POST https://13.94.148.187:4200/_sql (status: 401): retrying in 2s (19 left)
2021/12/21 12:02:16 [DEBUG] POST https://13.94.148.187:4200/_sql (status: 401): retrying in 4s (18 left)
2021/12/21 12:02:20 [DEBUG] POST https://13.94.148.187:4200/_sql (status: 401): retrying in 8s (17 left)
2021/12/21 12:02:28 [DEBUG] POST https://13.94.148.187:4200/_sql (status: 401): retrying in 16s (16 left)
2021/12/21 12:02:44 [DEBUG] POST https://13.94.148.187:4200/_sql (status: 401): retrying in 30s (15 left)
2021/12/21 12:03:14 [DEBUG] POST https://13.94.148.187:4200/_sql (status: 401): retrying in 30s (14 left)
2021/12/21 12:03:44 [DEBUG] POST https://13.94.148.187:4200/_sql (status: 401): retrying in 30s (13 left)
```

On top of those settings, Terratest will kill a test after [10 minutes](https://terratest.gruntwork.io/docs/testing-best-practices/timeouts-and-logging/) (increased in `run.sh` to 15 minutes).

File: test/run.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+source .env
+
+AWS_TEST_VPC_ID=$AWS_TEST_VPC_ID \
+AWS_TEST_SSH_KEYPAIR=$AWS_TEST_SSH_KEYPAIR \
+AWS_TEST_SUBNET_IDS=$AWS_TEST_SUBNET_IDS \
+AWS_TEST_AVAILABILITY_ZONES=$AWS_TEST_AVAILABILITY_ZONES \
+AZURE_TEST_SUBSCRIPTION_ID=$AZURE_TEST_SUBSCRIPTION_ID \
+go test -timeout 15m
```
Comment (jayeff) [inline test/run.sh:10]: Is it necessary to explicitly pass all the env variables to `go test`? 🤔 I would expect them to be loaded into environment by the `source` cmd and then to be available to go with `os.Getenv`

**Review by jayeff — APPROVED** [review]
one small note regarding env passing, other than that LGTM

File: test/run.sh (hunk 2 above)
Comment (hammerhead) [reply test/run.sh:10]: Yes, unfortunately, it is required to pass the environment variables explicitly. When using `source`, the imported environment variables are available in the very same process only. When invoking a sub-process (like `go` here), they aren't automatically passed.
Another approach can be to use `export` which will pass on environment variables also to any subsequent sub-processes, but therefore also giving less fine-grained control over exposing them.

Comment (jayeff) [reply test/run.sh:10]: I see. yes, would not work without `export` as you stated.

One alternative could be to use joho/godotenv but likely this small convenient factor does not justify adding an additional library.

**Pull request merged**